app/models/host/base.rb

```diff
@@ -370,5 +371,23 @@ def check_provision_interface
         errors.add :interfaces, _("managed host must have one provision interface")
       end
     end
+
+    # we can't use standard unique validation on interface since we can't properly handle :scope => :host_id
+    # for new hosts host_id does not exist at that moment, validation would work only for persisted records
+    def uniq_interfaces_identifiers
+      success = true
+      identifiers = []
+      self.interfaces.each do |interface|
+        next if interface.identifier.blank?
+        if identifiers.include?(interface.identifier)
```

Comment:
that's too smart :stuck_out_tongue_winking_eye: 
in code, follow the keep it simple principle...